rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestBookKeeper.java

```diff
@@ -138,4 +142,85 @@ public void testGetPrestoClusterManagerInValidInstance() throws Exception
+
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    //Rewrite the file with half the data
+    DataGen.populateFile(backendFilePath.toString(), 2);
+
+    expectedFileSize = DataGen.generateContent(2).length();
+
+    info = bookKeeper.getFileInfo(backendFilePath.toString());
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+  }
+
+  @Test
+  public void testGetFileInfoWithInvalidationDisabled() throws Exception
+  {
+    Path backendFilePath = new Path(BookKeeperTestUtils.getDefaultTestDirectoryPath(conf), BACKEND_FILE_NAME);
+    DataGen.populateFile(backendFilePath.toString());
+    int expectedFileSize = DataGen.generateContent(1).length();
+
+    CacheConfig.setFileInvalidationEnabled(conf, false);
+
+    BookKeeper bookKeeper = new CoordinatorBookKeeper(conf, metrics);
+    FileInfo info = bookKeeper.getFileInfo(backendFilePath.toString());
+
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    //Rewrite the file with half the data
+    DataGen.populateFile(backendFilePath.toString(), 2);
+
+    info = bookKeeper.getFileInfo(backendFilePath.toString());
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+  }
+
+  @Test
+  public void testGetFileInfoWithInvalidationDisabledWithCacheExpired() throws Exception
+  {
+    Path backendFilePath = new Path(BookKeeperTestUtils.getDefaultTestDirectoryPath(conf), BACKEND_FILE_NAME);
+    DataGen.populateFile(backendFilePath.toString());
+    int expectedFileSize = DataGen.generateContent(1).length();
+
+    CacheConfig.setFileInvalidationEnabled(conf, false);
+    CacheConfig.setFileStatusExpiryPeriod(conf, 5000);
+
+    BookKeeper bookKeeper = new CoordinatorBookKeeper(conf, metrics);
+    FileInfo info = bookKeeper.getFileInfo(backendFilePath.toString());
+
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    //Rewrite the file with half the data
+    DataGen.populateFile(backendFilePath.toString(), 2);
+
+    info = bookKeeper.getFileInfo(backendFilePath.toString());
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    // Sleep for 5 sec to expire the cache
+    Thread.sleep(5000);
```

Comment:
You should use ticks here. Guava cache has a way to do that. There must be an example of that written by @jordanw-bq

Reply:
@abhishekdas99 https://github.com/qubole/rubix/blob/6ebb35c2d9c44a48f2643f296a620fc9ea21bd85/rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestCoordinatorBookKeeper.java#L100

This test uses a `FakeTicker` to advance the cache in order to expire the entries. I had to implement a Mock(Coordinator)BookKeeper in order to supply a custom ticker, though (L123 of the same test class)